test/jdk/java/net/Socket/B8312065.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023, Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8312065
+ * @summary Socket.connect does not timeout as expected when profiling (i.e. keep receiving signal)
+ * @requires (os.family != "windows")
+ * @compile NativeThread.java
+ * @run main/othervm/native/timeout=120 -Djdk.net.usePlainSocketImpl B8312065
+ */
+
+import sun.misc.Signal;
+
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+
+public class B8312065 {
+    public static void main(String[] args) throws Exception {
+        System.loadLibrary("NativeThread");
+
+        // Setup SIGPIPE handler
+        Signal.handle(new Signal("PIPE"), System.out::println);
+
+        long osThreadId = NativeThread.getID();
+
+        Thread t = new Thread(() -> {
+            try {
+                // Send SIGPIPE to the thread every second
+                for (int i = 0; i < 10; i++) {
+                    if (NativeThread.signal(osThreadId, NativeThread.SIGPIPE) != 0) {
+                        System.out.println("Failed to send signal");
+                        System.exit(1);
+                    }
+                    Thread.sleep(1000);
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            } finally {
+                System.out.println("Test FAILED");
```

Comment:
don't you think "Test FAILED" will print always ?

Reply:
Hi vyommani. There are two situations in which this sentence is executed, one is that it loops for 10 seconds (this will not happen on the JDK I repaired), and the other is that the sleep is interrupted (it will not happen).
I ran the test on the JDK containing my fix and watched stdout, the message was indeed not printed and the test passed. On the JDK before the repair, it will print, and the test will fail.
But in order to make this code look clearer, I removed 'finnaly'.